ocaml/xapi/xapi_session.ml

```diff
@@ -398,19 +398,27 @@ let is_subject_suspended ~__context ~cache subject_identifier =
     debug "Subject identifier %s is suspended" subject_identifier ;
   (is_suspended, subject_name)
 
+let reusable_pool_session = ref Ref.null
+
+let reusable_pool_session_lock = Mutex.create ()
+
 let destroy_db_session ~__context ~self =
   Context.with_tracing ~__context __FUNCTION__ @@ fun __context ->
-  Xapi_event.on_session_deleted self ;
-  (* unregister from the event system *)
-  (* This info line is important for tracking, auditability and client accountability purposes on XenServer *)
-  (* Never print the session id nor uuid: they are secret values that should be known only to the user that *)
-  (* logged in. Instead, we print a non-invertible hash as the tracking id for the session id *)
-  (* see also task creation in context.ml *)
-  (* CP-982: create tracking id in log files to link username to actions *)
-  info "Session.destroy %s" (trackid self) ;
-  Rbac_audit.session_destroy ~__context ~session_id:self ;
-  (try Db.Session.destroy ~__context ~self with _ -> ()) ;
-  Rbac.destroy_session_permissions_tbl ~session_id:self
+  with_lock reusable_pool_session_lock (fun () ->
+      if self <> !reusable_pool_session then (
+        Xapi_event.on_session_deleted self ;
+        (* unregister from the event system *)
+        (* This info line is important for tracking, auditability and client accountability purposes on XenServer *)
+        (* Never print the session id nor uuid: they are secret values that should be known only to the user that *)
+        (* logged in. Instead, we print a non-invertible hash as the tracking id for the session id *)
+        (* see also task creation in context.ml *)
+        (* CP-982: create tracking id in log files to link username to actions *)
+        info "Session.destroy %s" (trackid self) ;
+        Rbac_audit.session_destroy ~__context ~session_id:self ;
+        (try Db.Session.destroy ~__context ~self with _ -> ()) ;
+        Rbac.destroy_session_permissions_tbl ~session_id:self
+      )
```

Comment:
Let's log the else-case here, so that it is clear why there is no `Session.destroy` in the logs as part of a `session.logout` call.